th2_data_services/provider/v5/commands/http.py

```diff
@@ -11,23 +11,273 @@
+                the one closest to the specified timestamp.
+            limit_for_parent: How many children events for each parent do we want to request.
+            metadata_only: Receive only metadata (true) or entire event (false) (without attached_messages).
+            attached_messages: Gets messages ids which linked to events.
+            filters: Filters using in search for messages.
+
+        """
+        self._start_timestamp = start_timestamp
+        self._end_timestamp = end_timestamp
+        self._parent_event = parent_event
+        self._search_direction = search_direction
+        self._resume_from_id = resume_from_id
+        self._result_count_limit = result_count_limit
+        self._keep_open = keep_open
+        self._limit_for_parent = limit_for_parent
+        self._metadata_only = metadata_only
+        self._attached_messages = attached_messages
+        self._filters = filters
+
+        self._adapters = AdapterSSE()
+
+    def handle(self, data_source: HTTPProvider5DataSource):
+        api: HTTPProvider5API = data_source.source_api
+        url = api.get_url_search_sse_events(
+            start_timestamp=self._start_timestamp,
+            end_timestamp=self._end_timestamp,
+            parent_event=self._parent_event,
+            search_direction=self._search_direction,
+            resume_from_id=self._resume_from_id,
+            result_count_limit=self._result_count_limit,
+            keep_open=self._keep_open,
+            limit_for_parent=self._limit_for_parent,
+            metadata_only=self._metadata_only,
+            attached_messages=self._attached_messages,
+            filters=self._filters,
+        )
+
+        logger.info(url)
+
+        return Data(partial(api.execute_sse_request, url)).map(self._adapters.handle)
+
+
+class GetMessageById(IHTTPProvider5Command):
+    """A Class-Command for request to rpt-data-provider.
+
+    It retrieves the message by id.
+
+    Returns:
+        dict: Th2 message.
+    """
+
+    def __init__(self, id: str):
+        """
+        Args:
+            id: Message id.
+
+        """
+        self._id = id
+
+    def handle(self, data_source: HTTPProvider5DataSource):
+        api: HTTPProvider5API = data_source.source_api
+        url = api.get_url_find_message_by_id(self._id)
+
+        logger.info(url)
+
+        return api.execute_request(url).json()
+
+
+class GetMessagesById(IHTTPProvider5Command):
+    """A Class-Command for request to rpt-data-provider.
+
+    It retrieves the messages by ids.
+
+    Returns:
+        List[dict]: Th2 messages.
+    """
+
     def __init__(self, ids: List[str]):
-        self._ids: ids
+        """
+        Args:
+            ids: Message id list.
+
+        """
+        self._ids: ids = ids
+
+    def handle(self, data_source: HTTPProvider5DataSource):
+        api: HTTPProvider5API = data_source.source_api
+        result = []
+        for id in self._ids:
+            url = api.get_url_find_message_by_id(id)
```

Comment:
Use `GetMessageById` instead because you already realize a single message getting. 
It's better to avoid duplicates.